packages/clay-date-picker/src/ClayDatePicker.js

```diff
@@ -0,0 +1,384 @@
+	 * @type {!Array<Array<{date: Date, ?outside: bool}>>}
+	 */
+	const [weeks, setWeeks] = useWeeks(currentMonth, firstDayOfWeek);
+
+	/**
+	 * Flag to indicate if date is expanded.
+	 * @type {?bool}
+	 */
+	const [expanded, setExpanded] = useState(initialExpanded);
+
+	/**
+	 * Create a ref to store the datepicker DOM element
+	 */
+	const elementRef = useRef(null);
+
+	/**
+	 * Handles the click on element of the day
+	 * @param {!Date} date
+	 */
+	function handleDayClicked(date) {
+		setDaySelected(date);
+		setInputValue(date);
+	}
+
+	/**
+	 * Handles the input change.
+	 * @param {!Event} event
+	 */
+	function inputChange(event) {
+		const {value} = event.target;
+		const format = time ? `${dateFormat} ${time}` : dateFormat;
+		const date = moment(value, format);
+
+		if (date.isValid()) {
+			setCurrentMonth(date.toDate());
+			setDaySelected(date.toDate());
+
+			if (time) {
+				setCurrentTime(date.hours(), date.minutes());
+			}
+		}
+
+		if (onValueChange) {
+			onValueChange(value);
+		}
+
+		setInputValue(value);
+	}
+
+	/**
+	 * Handles datepicker view
+	 */
+	function handleCalendarButtonClicked() {
+		setExpanded(!expanded);
+	}
+
+	useEffect(
+		() => {
+			// It is not necessary to update the weeks when the native
+			// date is enabled. Unnecessary rendering and processing.
+			if (!useNative) {
+				setWeeks(currentMonth, firstDayOfWeek);
+			}
+		},
+		[currentMonth]
+	);
+
+	const classNames = classnames('input-group', {
+		[id]: id,
+	});
+
+	return (
+		<div className={classNames} ref={elementRef}>
+			<div className="input-group-item">
+				<InputDate
+					ariaLabel={ariaLabel}
+					currentTime={currentTime}
+					dateFormat={dateFormat}
+					defaultValue={defaultValue}
+					inputName={inputName}
+					onChange={inputChange}
+					placeholder={placeholder}
+					time={time}
+					timeFormat={timeFormat}
+					useNative={useNative}
+					value={inputValue}
+				/>
+				{!useNative && (
+					<div className="input-group-inset-item input-group-inset-item-after">
+						<Button
+							className="d-none d-md-inline-block"
+							icon="calendar"
+							onClick={handleCalendarButtonClicked}
+							spritemap={spritemap}
+							style="unstyled"
+						/>
+					</div>
+				)}
+			</div>
+			{!useNative && (
+				<Dropdown
+					containerRef={elementRef}
+					expanded={expanded}
+					onDocumentClick={() => setExpanded(false)}
+				>
+					<React.Fragment>
+						<div className="datepicker">
+							<DateNavigation
+								currentMonth={currentMonth}
+								months={months}
+								onDotClicked={() => {
+									setCurrentMonth(initialMonth);
+									setDaySelected(initialMonth);
+									setInputValue(initialMonth);
+								}}
+								onMonthChange={setCurrentMonth}
+								spritemap={spritemap}
+								years={years}
+							/>
+							<Weekdays
+								firstDayOfWeek={firstDayOfWeek}
+								weekdaysShort={weekdaysShort}
+							>
+								{({weekday, key}) => (
```

Comment:
You may want to add this plugin to catch sorting destructured keys. https://github.com/mthadley/eslint-plugin-sort-destructure-keys

Reply:
Thanks for the tip!